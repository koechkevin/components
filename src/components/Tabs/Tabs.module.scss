@import '../../styles/global';

.tabs {
  :global {
    .ant-tabs-bar {
      border-bottom: none;
    }

    .ant-tabs-nav .ant-tabs-tab-active {
      color: $color-primary !important;
      font-size: 12px;
      font-weight: 600;
      line-height: 18px;
    }

    .ant-tabs-nav .ant-tabs-tab {
      color: $color-base-dark-gray;
      font-size: 12px;
      font-weight: 600;
      text-align: center;
      line-height: 18px;
    }

    .ant-tabs-tab {
      margin: 0px;
      padding: 15px 21px;
    }

    .ant-tabs-nav {
      color: $color-hex-gray-primary;
      text-align: center;
    }

    .ant-tabs-ink-bar {
      height: 2px;
      background: $color-primary;
    }

    .ant-tabs-nav .ant-tabs-tab {
      border-bottom: 3px solid transparent;

      &:hover {
        color: $color-base-dark-gray;
        border-bottom: 3px solid $color-primary-light;
      }
    }

    .ant-tabs-left-bar {
      .ant-tabs-nav .ant-tabs-tab {
        color: $color-base-dark-gray;
        border-bottom: none !important;
        border-right: 3px solid transparent !important;

        &:hover {
          color: $color-base-dark-gray;
          border-bottom: none !important;
          border-right: 3px solid $color-primary-light !important;
        }
      }
    }
  }
}
