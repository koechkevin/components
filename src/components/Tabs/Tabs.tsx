import { Tabs as AntTabs } from 'antd';
import { TabsProps } from 'antd/lib/tabs';
import React, { FC } from 'react';

import styles from './Tabs.module.scss';

const Tabs: FC<TabsProps> = (props) => {
  const { children, className, ...restProps } = props;
  const classNames: string = [styles.tabs, className].join(' ');

  return (
    <AntTabs className={classNames} {...restProps} animated={{ tabPane: false, inkBar: true }}>
      {children}
    </AntTabs>
  );
};

export default Tabs;
