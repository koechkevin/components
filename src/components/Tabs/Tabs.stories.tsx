import { faFolderOpen, faHeart, faStar } from '@fortawesome/pro-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { storiesOf } from '@storybook/react';
import { Tabs as AntdTabs } from 'antd';
import React from 'react';

import { Center } from '../Center';
import Tabs from './Tabs';

const { TabPane } = AntdTabs;
const stories = storiesOf('Tabs', module);

stories.addDecorator((story) => <Center>{story()}</Center>);

stories
  .add(
    'Tabs',
    () => (
      <Tabs>
        <TabPane tab="Tab 1" key="1">
          Content of Tab Pane 1
        </TabPane>
        <TabPane tab="Tab 2" key="2">
          Content of Tab Pane 2
        </TabPane>
        <TabPane tab="Tab 3" key="3">
          Content of Tab Pane 3
        </TabPane>
      </Tabs>
    ),
    {
      notes: 'Tabs library component',
    },
  )
  .add(
    'Tabs with icons',
    () => (
      <Tabs>
        <TabPane
          tab={
            <span>
              <FontAwesomeIcon icon={faHeart} style={{ marginRight: 9 }} />
              Tab 1
            </span>
          }
          key="1"
        >
          Content of Tab Pane 1
        </TabPane>
        <TabPane
          tab={
            <span>
              <FontAwesomeIcon icon={faStar} style={{ marginRight: 9 }} />
              Tab 2
            </span>
          }
          key="2"
        >
          Content of Tab Pane 2
        </TabPane>
        <TabPane
          tab={
            <span>
              <FontAwesomeIcon icon={faFolderOpen} style={{ marginRight: 9 }} />
              Tab 3
            </span>
          }
          key="3"
        >
          Content of Tab Pane 3
        </TabPane>
      </Tabs>
    ),
    {
      notes: 'Tabs library component',
    },
  )
  .add(
    'Vertical position',
    () => (
      <Tabs tabPosition="left">
        <TabPane tab="Tab 1" key="1">
          Content of Tab Pane 1
        </TabPane>
        <TabPane tab={'Tab 2'} key="2">
          Content of Tab Pane 2
        </TabPane>
        <TabPane tab="Tab 3" key="3">
          Content of Tab Pane 3
        </TabPane>
      </Tabs>
    ),
    {
      notes: 'Tabs library component',
    },
  );
