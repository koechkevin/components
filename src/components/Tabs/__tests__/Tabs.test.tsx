import { Tabs as AntdTabs } from 'antd';
import { mount } from 'enzyme';
import React from 'react';

import Tabs from '../Tabs';

const { TabPane } = AntdTabs;

describe('Tests for Tabs component', () => {
  const wrapper = mount(
    <Tabs>
      <TabPane tab="Tab 1" key="1">
        Content of Tab Pane 1
      </TabPane>
      <TabPane tab="Tab 2" key="2">
        Content of Tab Pane 2
      </TabPane>
      <TabPane tab="Tab 3" key="3">
        Content of Tab Pane 3
      </TabPane>
    </Tabs>,
  );

  it('should render Tabs without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Tabs style class', () => {
    expect(wrapper.find('.tabs').exists()).toEqual(true);
  });
});
