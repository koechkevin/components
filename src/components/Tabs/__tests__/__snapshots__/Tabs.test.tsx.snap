// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Tabs component should render Tabs without crashing 1`] = `ReactWrapper {}`;
